zhmccli/_cmd_user_pattern.py

```diff
@@ -0,0 +1,674 @@
+              # HMC property: 'domain-name-restrictions'
+              help='Name of the LDAP attribute on the LDAP server that '
+              'is used for lookup of the domain name restrictions, which '
+              'specifies the domain name restrictions (i.e. to which consoles '
+              'the pattern-based user may log on to). '
+              'Default: There are no such restrictions. '
+              'HMC property: domain-name-restrictions')
+@click.pass_obj
+def user_pattern_create(cmd_ctx, **options):
+    """
+    Create an HMC user pattern.
+
+    The user template to be used when creating the pattern-based user for
+    the LDAP user that logs on must be identified using one of these methods.
+    The properties related to the methods that are not used, will be reset:
+
+    \b
+    * directly by specifying a user template name (--specific-template option)
+    * via LDAP attribute lookup (--ldap-template-... options)
+    * via LDAP group lookup (--ldap-group-template-... options)
+
+    For some options, the corresponding HMC properties are different than
+    the option names. In such cases, the option descriptions mention the
+    corresponding HMC property names.
+
+    There are some old properties that are set by the HMC consistent with
+    the preferred new properties. Because of that, the old properties have
+    no corresponding options and the options for the new preferred properties
+    should be used instead. The old properties are:
+
+    \b
+    * ldap-server-definition-uri:
+      Name of LDAP Server Definition object used for LDAP attribute lookup for
+      specific user template name and domain name restrictions. Use the
+      --ldap-template-lsd and --domain-name-restrictions-lsd options instead.
+    * user-template-uri:
+      Name of the specific user template to be used. Use the
+      --specific-template option instead.
+
+    In addition to the command-specific options shown in this help text, the
+    general options (see 'zhmc --help') can also be specified right after the
+    'zhmc' command name.
+    """
+    cmd_ctx.execute_cmd(lambda: cmd_user_pattern_create(cmd_ctx, options))
+
+
+@user_pattern_group.command('update', options_metavar=COMMAND_OPTIONS_METAVAR)
+@click.argument('USER_PATTERN', type=str, metavar='USER_PATTERN')
+@click.option('--name', type=str, required=False,
+              help='The new name of the user pattern.')
+@click.option('--description', type=str, required=False,
+              help='The new description of the user pattern.')
+@click.option('--pattern', type=str, required=False,
+              help='The new pattern for matching userids. '
+              'The combination of "pattern" and "type" must be unique.')
+@click.option('--type', type=click.Choice(['glob-like', 'regular-expression']),
+              required=False,
+              help='The new style in which the pattern is expressed.')
+@click.option('--retention-time', type=int, required=False, default=1,
```

Comment:
same here

Reply:
On update, there should not be a default value specified, because the default should be not to change the value. I'll change that.,

Reply:
DONE